src/components/molecules/RegisterReferralForm.tsx

```diff
@@ -0,0 +1,51 @@
+import {useTranslation} from "react-i18next";
+import AppNextButton from "../atom/AppNextButton";
+import CrossIcon from "../icons/CrossIcon";
+import UploadImageIcon from "../icons/UploadImageIcon";
+import useRegisterReferralForm from "../../hooks/useRegisterReferralForm";
+import {namespaces} from "../../i18n/i18n.constants";
+
+const RegisterReferralForm = () => {
+    const {file, setFile, inputRef, cameraRef, handleFileChange, handleClick, handleOnClick} = useRegisterReferralForm();
+    const {t} = useTranslation(namespaces.pages.registerReferral);
+
+    return (
+        <div className="mt-8 flex h-full w-full flex-col justify-between gap-4 self-center md:w-1/2 lg:w-1/3">
+            <div className="flex flex-col gap-4 rounded-lg bg-white p-4">
+                <div className="flex flex-col gap-1.5 text-primary-color">
+                    {!file && <div>
+                        <div className="flex h-52 flex-col place-content-center gap-5 lg:h-96">
+                            <UploadImageIcon/>
+                            <p className="text-center font-text">{t("title")}</p>
+                        </div>
+                        <div className="grid grid-cols-2 gap-4">
+                            <input type="file" id="file" className="hidden" onChange={handleFileChange} ref={inputRef}/>
+                            <button className="mt-4 rounded-2xl px-2 py-5 text-white font-button bg-secondary-color"
+                                    onClick={() => handleClick(inputRef)}>
+                                {t("upload")}
+                            </button>
+                            <input type="file" id="camera" accept="image/*" capture onChange={handleFileChange}
+                                   style={{display: "none"}} ref={cameraRef}/>
+                            <button className="mt-4 rounded-2xl px-2 py-5 text-white font-button bg-primary-color"
+                                    onClick={() => handleClick(cameraRef)}>
+                                {t("camera")}
+                            </button>
+                        </div>
+                    </div>}
```

Comment:
I usually divide this type of component into different components.
If it doesn't have a file, one component is needed, and another when it has a file

Reply:
Hi @yodra! I have completed the changes on this pull request as requested. I have refactored the RegisterReferralForm component into smaller components and refactored the AppNextButton to add a buttonType on it. Please let me know if there is anything else that needs to be implemented or if there are any improvements that can be made to the current changes. Thank you! 😄